src/wallet/wallet.h

```diff
@@ -679,6 +680,28 @@ struct CoinSelectionParams
 };
 
 class WalletRescanReserver; //forward declarations for ScanForWalletTransactions/RescanFromTime
+
+namespace bm = boost::multi_index;
+
+struct ByHash {};
+struct ByOrder {};
+struct ByTime {};
+
+typedef bm::multi_index_container<
+    CWalletTx,
+    bm::indexed_by<
+        bm::ordered_unique<
+            bm::tag<ByHash>, bm::const_mem_fun<CWalletTx, const uint256&, &CWalletTx::GetHash>
+        >,
+        bm::ordered_non_unique<
+            bm::tag<ByOrder>, bm::member<CWalletTx, int64_t, &CWalletTx::nOrderPos>
+        >,
+        bm::ordered_non_unique<
+            bm::tag<ByTime>, bm::member<CWalletTx, unsigned int, &CWalletTx::nTimeReceived>
+        >
+    >
+> mapWalletType;
```

Comment:
The type change to `multiIndexWalletType` will be more readable.

Reply:
Underlying type multi_index isn't needed, about me.